cloud/aws/templates/aws_oidc/app.tf

```diff
@@ -188,10 +145,166 @@ module "td" {
+      "awslogs-stream-prefix" = "ecs"
+      "awslogs-group"         = module.aws_scraper_logs.logs_path
+      "awslogs-create-group"  = "true"
+      # Use https://docs.docker.com/config/containers/logging/awslogs/#awslogs-multiline-pattern
+      # Logs are streamed via container's stdout. Each line is considered a
+      # separate log messsage. To collect stacktraces, which take multiple line,
+      # to a single event we consider all lines which start with a whitespace character to be
+      # part of the previous line and not a standalone event.
+      "awslogs-multiline-pattern" = "^[^\\s]"
+    }
+    secretOptions = null
+  }
+}
+
+locals {
+  name_prefix = "${var.app_prefix}-civiform"
+
   tags = {
     Name = "${var.app_prefix} Civiform EC2 Task Definition"
     Type = "Civiform EC2 Task Definition"
   }
+
+  civiform_ecs_task_execution_role_custom_policies = [
+    jsonencode(
+      {
+        "Version" : "2012-10-17",
+        "Statement" : [
+          {
+            "Effect" : "Allow",
+            "Action" : [
+              "secretsmanager:GetSecretValue"
+            ],
+            "Resource" : [
+              aws_secretsmanager_secret.postgres_username_secret.arn,
+              aws_secretsmanager_secret.postgres_password_secret.arn,
+              aws_secretsmanager_secret.app_secret_key_secret.arn,
+              aws_secretsmanager_secret.api_secret_salt_secret.arn,
+              aws_secretsmanager_secret.adfs_secret_secret.arn,
+              aws_secretsmanager_secret.adfs_client_id_secret.arn,
+              aws_secretsmanager_secret.applicant_oidc_client_secret_secret.arn,
+              aws_secretsmanager_secret.applicant_oidc_client_id_secret.arn,
+            ]
+          },
+          {
+            "Effect" : "Allow",
+            "Action" : [
+              "kms:Encrypt",
+              "kms:Decrypt",
+              "kms:ReEncrypt*",
+              "kms:GenerateDataKey*",
+              "kms:DescribeKey"
+            ],
+            "Resource" : [aws_kms_key.civiform_kms_key.arn, aws_kms_key.file_storage_key.arn]
+          },
+          {
+            "Effect" : "Allow",
+            "Action" : [
+              "s3:*"
+            ],
+            "Resource" : [
+              aws_s3_bucket.civiform_files_s3.arn,
+              "${aws_s3_bucket.civiform_files_s3.arn}/*",
+            ]
+          },
+          {
+            "Effect" : "Allow",
+            "Action" : [
+              "ses:*"
+            ],
+            "Resource" : [
+              for email in module.email_service : email.email_arn
+            ]
+          },
+          {
+            "Effect" : "Allow",
+            "Action" : [
+              "aps:RemoteWrite"
+            ],
+            "Resource" : "*"
+          }
+        ]
+      }
+    )
+  ]
+}
+
+resource "aws_iam_role" "civiform_ecs_task_execution_role" {
+  name               = "${local.name_prefix}-ecs-task-execution-role"
+  assume_role_policy = <<JSON
+{
```

Comment:
can you indent this, it looks weird.

Reply:
I think that will prepend whitespace in the resulting string, which makes for invalid JSON